geomesa-spark/geomesa-spark-jts/src/main/scala/org/locationtech/geomesa/spark/jts/udf/GeometricProcessingFunctions.scala

```diff
@@ -42,23 +40,37 @@ object SQLGeometryProcessingFunctions extends LazyLogging {
   }
 
   val ST_antimeridianSafeGeom: Geometry => Geometry = nullableUDF(geom => {
-    GeohashUtils.getInternationalDateLineSafeGeometry(geom) match {
-      case Success(g) => g
-      case Failure(e) =>
-        logger.warn(s"Error splitting geometry on anti-meridian for $geom", e)
-        geom
+    def degreesToTranslate(x: Double): Double = (((x + 180) / 360.0).floor * -360).toInt
+    if (geom.getEnvelopeInternal.getMinX < -180 || geom.getEnvelopeInternal.getMaxX > 180) {
+
+      geom.apply(new CoordinateSequenceFilter() {
```

Comment:
would it be better to create a new geometry, and not mutate the one that is passed in?